ocaml/src/alba_osd.ml

```diff
@@ -0,0 +1,281 @@
+        self # range
+          prio
+          ~first ~finc ~last
+          ~reverse ~max
+        >>= fun ((cnt, names), has_more) ->
+        Lwt_list.map_p
+          (fun name ->
+           let object_name = Slice.get_string_unsafe name in
+           alba_client # download_object_to_bytes
+                       ~namespace
+                       ~object_name
+                       ~consistent_read
+                       ~should_cache:true
+           >>= function
+           | None -> Lwt.fail_with "object missing in range entries"
+           | Some (v, mf) ->
+              Lwt.return (name, v, mf.Nsm_model.Manifest.checksum))
+          names >>= fun r ->
+        Lwt.return ((cnt, r), has_more)
+
+      method partial_get prio name object_slices =
+        alba_client # download_object_slices
+                    ~namespace
+                    ~object_name:(Slice.get_string_unsafe name)
+                    ~object_slices:(List.map
+                                      (fun (offset, len, dst, dstoff) ->
+                                       Int64.of_int offset, len, dst, dstoff)
+                                      object_slices)
+                    ~consistent_read
+                    ~fragment_statistics_cb:ignore >>= function
+        | None -> Lwt.return Osd.NotFound
+        | Some _ -> Lwt.return Osd.Success
+
+      method apply_sequence prio asserts updates =
+        let prepare_updates namespace_id =
+          Lwt_list.map_p
+            (function
+              | Osd.Update.Set (key, None) ->
+                 Lwt.return (Nsm_model.Update.DeleteObject (Slice.get_string_unsafe key))
+              | Osd.Update.Set (key, Some (value, cs, _)) ->
+                 Alba_client_upload.upload_object''
+                   (alba_client # nsm_host_access)
+                   (alba_client # osd_access)
+                   (alba_client # get_base_client # get_preset_cache # get)
+                   (alba_client # get_base_client # get_namespace_osds_info_cache)
+                   ~object_t0:0.
+                   ~timestamp:(Unix.gettimeofday ())
+                   ~namespace_id
+                   ~object_name:(Slice.get_string_unsafe key)
+                   ~object_reader:(let open Asd_protocol.Blob in
+                                   match value with
+                                   | Lwt_bytes s -> new Object_reader.bytes_reader s
+                                   | Bigslice s -> new Object_reader.bigstring_slice_reader s
+                                   | Bytes s -> new Object_reader.string_reader s
+                                   | Slice s -> new Object_reader.slice_reader s)
+                   ~checksum_o:(Some cs)
+                   ~object_id_hint:None
+                   ~fragment_cache:(alba_client # get_base_client # get_fragment_cache)
+                   ~cache_on_write:(alba_client # get_base_client # get_cache_on_read_write |> snd)
```

Comment:
If we're accessing this much (all the `alba_client # ... occurrences) from alba_client , 
only to hand it to alba_client_upload.upload_object'' there should probably be a method for this in alba_client.